src/util/encoders/hex_encoder.cc

```diff
@@ -14,35 +14,29 @@
  * limitations under the License.
  *
  */
-#include "util/encoders/hex_encoder.h"
-
-#include <cstring>
-
-namespace veles {
-namespace util {
-namespace encoders {
-
-QString HexEncoder::displayName(bool decode) {
-  if (decode) {
-    return "decode from raw hex";
-  }
-
-  return "encode to raw hex";
-}
-
-QString HexEncoder::encode(const QByteArray &data) {
-  QByteArray ba = data.toHex();
-  return QString::fromLatin1(ba);
-}
-
-QByteArray HexEncoder::decode(const QString &str) {
-  return QByteArray().fromHex(str.toLatin1());
-}
-
-bool HexEncoder::validateEncoded(const QString &str) {
-  return Encoder::validateEncoded(str.toLower());
-}
-
-}  // namespace encoders
-}  // namespace util
-}  // namespace veles
+#include "util/encoders/hex_encoder.h"
+
+namespace veles {
+namespace util {
+namespace encoders {
+
+QString HexEncoder::encodingDisplayName() {
+  return "Hex";
+}
+
+QString HexEncoder::decodingDisplayName() {
+  return "Hex";
+}
+
+QString HexEncoder::encode(const QByteArray& data) {
+  QByteArray ba = data.toHex();
+  return QString::fromLatin1(ba);
+}
+
+QByteArray HexEncoder::decode(const QString& str) {
+  return QByteArray().fromHex(str.toLatin1());
```

Comment:
`fromHex` is static method, so there is no need to create empty `QByteArray` object.